### PR TITLE
Sharepoint List CLI commands added to package

### PR DESCRIPTION
**Pull request by dadams39**
## Description
CLI commands were added to facilitate restore of `SharePoint.Lists` to M365

## Does this PR need a docs update or release note?

- [x] :clock1: Yes, but in a later PR

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* related to #1963<issue>

## Test Plan

- [x] :zap: Unit test

### CODE REVIEW

**Comment by dadams39** [comment]
CLI Incomplete:
`./corso restore sharepoint --backup e5421913-8fca-478a-82ae-903134e06357 --lists-item d73571eb56c5`
creates the following following error.
```
Error: Failed to run SharePoint restore: 2 errors occurred:
	* errors prevented the operation from processing
	* retrieving service data: no restore path given
```

This is a selector misalignment within the package.

File: src/cli/utils/sharepoint.go [hunk 1]
```diff
@@ -9,12 +9,16 @@ import (
 const (
 	LibraryItemFN = "library-item"
 	LibraryFN     = "library"
+	ListsItemFN   = "lists-item"
+	ListsFN       = "lists"
```
Comment (ryanfkeepers) [inline src/cli/utils/sharepoint.go:13]: The plural here is a little confusing.  Let's stick with just `list` instead.
```suggestion
	ListItemFN   = "list-item"
	ListFN       = "list"
```

File: src/cli/restore/sharepoint.go [hunk 2]
```diff
@@ -64,6 +66,11 @@ func addSharePointCommands(cmd *cobra.Command) *cobra.Command {
 			utils.LibraryItemFN, nil,
 			"Restore library items by file name or ID")
 
+		fs.StringSliceVar(
```
Comment (ryanfkeepers) [inline src/cli/restore/sharepoint.go:69]: Halfway there: you're missing the StringSliceVar for `listPaths`.

Incidentally, line 60 is a bug and should be `&libraryPaths`

Comment (dadams39) [reply src/cli/restore/sharepoint.go:69]: Updates within [4a311ac](https://github.com/alcionai/corso/pull/1972/commits/4a311ac7405e7197bddf40cc32ce3de5110b216c)

**Comment by dadams39** [comment]
Works as advertised:
```
/corso/src$ ./corso restore sharepoint --backup af166ce0-5ef6-426d-a683-4c1847111de4 --list-item 09cd70b15912
Connecting to repository:      9s done 
Discovered 1 items in backup af166ce0-5ef6-426d-a683-4c1847111de4 to restore   
Enumerating items in repository:      0s done 
Connecting to M365:      1s done 
Restoring data:         done 
Error: Failed to run SharePoint restore: 2 errors occurred:
	* errors prevented the operation from processing
	* restoring service data: category lists not supported
```

File: src/cli/restore/sharepoint.go [hunk 3]
```diff
@@ -64,6 +66,16 @@ func addSharePointCommands(cmd *cobra.Command) *cobra.Command {
...
+			&listPaths,
+			utils.ListFN, nil,
+			"Restore list items by SharePoint path")
```
Comment (ryanfkeepers) [inline src/cli/restore/sharepoint.go:72]: `by path` won't be very clear to the user.  Maybe this is better?
```suggestion
			"Restore list items by SharePoint list ID")
```

File: src/cli/restore/sharepoint.go [hunk 4]
```diff
@@ -64,6 +66,16 @@ func addSharePointCommands(cmd *cobra.Command) *cobra.Command {
...
+
+		fs.StringSliceVar(
+			&listItems,
+			utils.ListItemFN, nil,
+			"Restore list items by file name or ID")
```
Comment (ryanfkeepers) [inline src/cli/restore/sharepoint.go:77]: List items don't have file names, do they?
```suggestion
			"Restore list items by ID")
```

**Review by ryanfkeepers — APPROVED** [review]
Some nits, but generally looks good.

**Pull request merged**